File: duyo-backend/src/duyo/schemas/tamagochi.py
"""Tamagochi request/response schemas (Concept §4)."""

from typing import Literal

from pydantic import BaseModel, Field

# Which metric an interaction restores, and the allowed boost amount.
InteractionKind = Literal["check_in", "lesson", "play", "activity"]


class TamagochiRead(BaseModel):
    energy: int
    joy: int
    learning: int
    health: int

    model_config = {"from_attributes": True}


class TamagochiInteract(BaseModel):
    """A care interaction that restores one or more metrics (Concept §4.1).

    `kind` maps to the canonical restore (check_in→energy, lesson→learning,
    play→joy, activity→health). `amount` bounds how much a single interaction
    can add, so the client can't max out metrics in one call.
    """

    kind: InteractionKind
    amount: int = Field(default=10, ge=1, le=30)

File: duyo-mobile/src/api/endpoints/tamagochi.ts
import { apiClient } from '@/api/client';

// Mirrors backend duyo.schemas.tamagochi (Concept §4). Metrics are 0–100.

export interface TamagochiState {
  energy: number;
  joy: number;
  learning: number;
  health: number;
}

export type InteractionKind = 'check_in' | 'lesson' | 'play' | 'activity';

export interface InteractInput {
  kind: InteractionKind;
  amount?: number;
}

export async function getTamagochi(childId: string): Promise<TamagochiState> {
  const { data } = await apiClient.get<TamagochiState>(`/tamagochi/${childId}`);
  return data;
}

export async function interactTamagochi(
  childId: string,
  input: InteractInput,
): Promise<TamagochiState> {
  const { data } = await apiClient.post<TamagochiState>(
    `/tamagochi/${childId}/interact`,
    input,
  );
  return data;
}
